fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt

```diff
@@ -90,10 +92,26 @@ class Request(
         return this
     }
 
+    /**
+     *  <p> Note that your value, will be converted to a String via the toString() </p>
+     *  <p> Please note that header of the same key are supported and headers with the same key
+     *  will be sent in the format of `key` : [ value,value,value ] </p>
+     * @param pairs This all the key value pair you wish to add to the headers
+     *
+     * @return the request supplied
+     *
+     * */
     fun header(vararg pairs: Pair<String, Any>?): Request {
-        pairs.forEach {
-            if (it != null)
+        pairs.filterNotNull().forEach {
+            if (!headers.containsKey(it.first)) {
                 headers += Pair(it.first, it.second.toString())
+            } else {
+                headers[it.first] = headers[it.first]!!.let { value ->
```

Comment:
ah ... can we not use `!!` though?

Reply:
headers[it.first] returns a nullable, and since line #106 checks for that, this will never fail...
...except that it is absolutely not thread safe.

Reply:
I would say that `fun <K, V> Map<K, V>.getValue(key: K): V` is probably better if we dont want to deal with nullable value.

Reply:
Agreed!